enumerable_methods.rb

```diff
@@ -0,0 +1,145 @@
+# rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    while i < size
+      yield self[i]
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    while i < size
+      yield(self[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    selected_array = []
+    my_each do |element|
+      selected_array << element if yield(element)
+    end
+    selected_array
+  end
+
+  def my_all?(_para = nil)
+    result = true
+    if block_given?
+      my_each do |element|
+        result = false unless yield(element)
+      end
+    else
+      my_each do |element|
+        result = false unless element
+      end
+    end
+    result
+  end
```

Comment:
* [ ] should have working `my_all?`  identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

#### my_all? 

- [ ] when a class is passed as an argument returns true only if all of the collection is a member of such class, otherwise, it should return false

```ruby     
[1, 2.5, 'a', 9].my_all?(Integer) == [1, 2.5, 'a', 9].all?(Integer) => false
```
- Yours is retrieving the wrong answer
![image](https://user-images.githubusercontent.com/52209067/80103257-30646900-8531-11ea-8ceb-8da432809da6.png)

- Real all?
![image](https://user-images.githubusercontent.com/52209067/77688467-be1a5c00-6f65-11ea-8970-15c9e4f4913f.png)

- [ ] when a Regex is passed as an argument returns true if all of the collection matches the Regex

```ruby     
%w[dog door rod blade].my_all?(/o/) == %w[dog door rod blade].all?(/o/) => false
```
- Yours is retrieving the wrong answer
![image](https://user-images.githubusercontent.com/52209067/80103367-5e49ad80-8531-11ea-880b-c4cee8411117.png)

- Real all?
![image](https://user-images.githubusercontent.com/52209067/80103412-6d306000-8531-11ea-87b8-ad46000fc288.png)

- [ ] when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern

```ruby     
[3,4,7,11].my_all?(3) == [3,4,7,11].all?(3)  => false 
```
- Yours is retrieving the wrong answer
![image](https://user-images.githubusercontent.com/52209067/80103494-8d601f00-8531-11ea-94e9-68e501bf9dbd.png)

- Real all?
![image](https://user-images.githubusercontent.com/52209067/77689001-924ba600-6f66-11ea-9b72-210b3bb1d78b.png)